root/configure-dc-and-ca/Install-DC-and-CA.ps1

```diff
@@ -138,6 +138,44 @@ configuration CreateDCCA
             DependsOn = '[WindowsFeature]ADCS-Web-Enrollment','[xADCSCertificationAuthority]ADCS'
         }
 
+        Script Set_Admin_Password_Permanent
+        {
+            DependsOn  = @("[xADCSWebEnrollment]CertSrv")
+            GetScript  = { @{ Result = "Set_Admin_Password_Permanent"}}
+
+            TestScript = {
+                Test-Path -Path "C:\adminUpdate" #If file isn't found, then run SetScript.
+                }
+
+            SetScript  = {
+                function Set-Password-Permanent($uName){
+                    if(-not (Get-Password-Permanent -uName $uName)){
+                        Set-ADUser -Identity $uName -PasswordNeverExpires $true
+                        return Get-Password-Permanent -uName $uName
+                    }else{
+                        return $true
+                    }                
+                }
+
+                function Get-Password-Permanent($uName){
+                    $uInfo = Get-ADUser -Identity $uName -Properties PasswordNeverExpires
+                    return $uInfo.PasswordNeverExpires
+                }
+
+                $retry = 3
+                while(-not (Set-Password-Permanent -uName $($using:Admincreds.Username))){
+                    if(($retry--) -le 0){
+                        Write-Host "Failure to connect to Azure, Admin password not set to permanent."
+                    }
+                    Start-Sleep -seconds 30 # Wait for another attempt
+                }
+
+                $fileTest = New-Item "C:\adminUpdate" -type File
+                Set-Content -Path $fileTest -Value "Admin password permanent."
```

Comment:
This gets set even if it fails?

Reply:
New update should fix this. Should only call when PasswordNeverExpires is set to true. Re-organized a bit to correct for some bad logic handling failed connections.